qutip/tests/test_operators.py

```diff
@@ -196,30 +199,23 @@ def test_displace():
           0.j, -0.41083747 + 0.j],
          [0.00626025 + 0.j, 0.07418172 + 0.j, 0.41083747 + 0.j,
           0.90866411 + 0.j]])
-
-    assert_equal(np.allclose(dp.full(), dpmatrix), True)
+    np.testing.assert_allclose(dp.full(), dpmatrix, atol=1e-8)
 
 
-def test_charge():
-    "Charge operator"
-    N = 5
-    M = - np.random.randint(N)
-    ch = charge(N,M)
-    ch_matrix = np.diag(np.arange(M,N+1))
-    assert_equal(np.allclose(ch.full(), ch_matrix), True)
+@pytest.mark.parametrize("shift", [1, 2])
+def test_tunneling(shift):
+    N = 10
+    tn = qutip.tunneling(N, shift)
+    tn_matrix = np.diag(np.ones(N - shift), k=shift)
+    tn_matrix += tn_matrix.T
+    np.testing.assert_allclose(tn.full(), tn_matrix)
 
 
-def test_tunneling():
-    "Tunneling operator"
-    N = 5
-    tn = tunneling(2*N+1)
-    tn_matrix = np.diag(np.ones(2*N),k=-1) + np.diag(np.ones(2*N),k=1)
-    assert_equal(np.allclose(tn.full(), tn_matrix), True) 
-    
-    tn = tunneling(2*N+1,2)
-    tn_matrix = np.diag(np.ones(2*N-1),k=-2) + np.diag(np.ones(2*N-1),k=2)
-    assert_equal(np.allclose(tn.full(), tn_matrix), True)
-
+def test_commutator():
+    A = qutip.qeye(N)
+    B = qutip.destroy(N)
+    assert qutip.commutator(A, B) == qutip.qzero(N)
```

Comment:
we should test `commutator(type='anti')` as well.  I'm not sure anyone is actually using the function given how long it is to type compared to how simple it is conceptually, but still!